lib/matplotlib/rcsetup.py

```diff
@@ -176,20 +176,29 @@ def validate_bool_maybe_none(b):
 
 
 def _validate_date_converter(s):
+    if s is None:
+        return
     s = validate_string(s)
-    mdates = sys.modules.get("matplotlib.dates")
-    if mdates:
+    if s not in ['auto', 'concise']:
+        cbook._warn_external(f'date.converter string must be "auto" '
+                             f'or "concise", not "{s}".  Check your '
+                             'matplotlibrc')
+    try:
+        import matplotlib.dates as mdates
```

Comment:
What's the case where this import would fail?

Reply:
I'm not sure the import can fail, but I think it did with some sort of circular import error at some point when developing this.  Lets try w/o.